.github/workflows/clear-cache.yml

```diff
@@ -0,0 +1,35 @@
+name: Clear Gradle Cache
+
+on:
+  workflow_dispatch:
+  schedule:
+    # Every nights at 4
+    - cron: "0 4 * * *"
+
+# Enrich gradle.properties for CI/CD
+env:
+  GRADLE_OPTS: -Dorg.gradle.jvmargs="-Xmx8g -Dfile.encoding=UTF-8 -XX:+HeapDumpOnOutOfMemoryError" -Dkotlin.incremental=false -XX:+UseParallelGC
+  CI_GRADLE_ARG_PROPERTIES: --stacktrace -PpreDexEnable=false --max-workers 8 --warn
+
+jobs:
+  tests:
+    name: Clear Gradle cache
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: ⏬ Checkout with LFS
+        uses: nschloe/action-cached-lfs-checkout@v1.2.2
+      - name: ☕️ Use JDK 17
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin' # See 'Supported distributions' for available options
+          java-version: '17'
+      - name: Configure gradle
+        uses: gradle/actions/setup-gradle@v3
+        with:
+          cache-read-only: ${{ github.ref != 'refs/heads/develop' }}
```

Comment:
Do we really need this check?
I am wondering if it can block the action if ever we run it manually to another branch than develop (which should not happen but...). We will have `cache-read-only: true` and `gradle-home-cache-cleanup: true`. One parameter is probably taking precedence on the other, but we can also maybe just remove this line, WDYT?